src/ios/SimpleCameraPreview.m

```diff
@@ -65,28 +65,32 @@ - (void) enable:(CDVInvokedUrlCommand*)command {
     // Setup session
     self.sessionManager.delegate = self.cameraRenderController;
     
-    NSDictionary *setupSessionOptions;
+    NSMutableDictionary *setupSessionOptions = [NSMutableDictionary dictionary];
     if (command.arguments.count > 0) {
         NSDictionary* config = command.arguments[0];
         @try {
             if (config[@"targetSize"] != [NSNull null] && ![config[@"targetSize"] isEqual: @"null"]) {
                 NSInteger targetSize = ((NSNumber*)config[@"targetSize"]).intValue;
-                setupSessionOptions = @{ @"targetSize" : [NSNumber numberWithInteger:targetSize] };
+                [setupSessionOptions setValue:[NSNumber numberWithInteger:targetSize] forKey:@"targetSize"];
+            }
+            NSString *captureDevice = config[@"captureDevice"];
+            if (captureDevice != [NSNull null] && ![captureDevice isEqual: @""]) {
+                [setupSessionOptions setValue:captureDevice forKey:@"captureDevice"];
             }
         } @catch(NSException *exception) {
             [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"targetSize not well defined"] callbackId:command.callbackId];
         }
     }
     
     self.photoSettings = [AVCapturePhotoSettings photoSettingsWithFormat:@{AVVideoCodecKey : AVVideoCodecTypeJPEG}];
-    
+    NSDictionary *immutableSetupSessionOptions = [setupSessionOptions copy];
     [self.sessionManager setupSession:@"back" completion:^(BOOL started) {
         dispatch_async(dispatch_get_main_queue(), ^{
             CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
             [pluginResult setKeepCallbackAsBool:true];
             [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
         });
-    } options:setupSessionOptions photoSettings:self.photoSettings];
+    } options:immutableSetupSessionOptions photoSettings:self.photoSettings];
```

Comment:
Is there a reason why we need to make a copy of the options. I don't see that we are doing any manipulation with the `immutableSetupSessionOptions`